tests/istanbul/eip152_blake2/test_blake2.py

```diff
@@ -523,50 +555,412 @@ class ExpectedOutput(TestParameterGroup):
 )
 def test_blake2b(
     state_test: StateTestFiller,
-    data: Blake2bInput,
-    output: ExpectedOutput,
     pre: Alloc,
+    blake2b_contract_bytecode: Bytecode,
+    data: Union[Blake2bInput, str, bytes],
+    output: ExpectedOutput,
 ):
     """Test BLAKE2b precompile."""
     env = Environment()
 
-    account = pre.deploy_contract(
-        # Store all CALLDATA into memory (offset 0)
-        Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
-        # Store the returned CALL status (success = 1, fail = 0) into slot 0:
-        + Op.SSTORE(
-            0,
-            # Setup stack to CALL into Blake2b with the CALLDATA and CALL into it (+ pop value)
-            Op.CALL(
-                address=9,
-                args_offset=0,
-                args_size=Op.CALLDATASIZE(),
-                ret_offset=0x200,
-                ret_size=0x40,
-                gas_limit=1_000_000,
-            ),
-        )
-        + Op.SSTORE(
-            1,
-            Op.MLOAD(0x200),
+    account = pre.deploy_contract(blake2b_contract_bytecode, storage={0: 0xDEADBEEF})
+    sender = pre.fund_eoa()
+
+    if isinstance(data, Blake2bInput):
+        data = data.create_blake2b_tx_data()
+    elif isinstance(data, str):
+        data = bytes.fromhex(data)
+
+    tx = Transaction(
+        ty=0x0,
+        to=account,
+        data=data,
+        gas_limit=1_000_000,
+        protected=True,
+        sender=sender,
+        value=100000,
+    )
+
+    post = {
+        account: Account(
+            storage={
+                0: 0x1 if output.call_succeeds else 0x0,
+                1: output.data_1,
+                2: output.data_2,
+            }
         )
-        + Op.SSTORE(
-            2,
-            Op.MLOAD(0x220),
+    }
+    state_test(env=env, pre=pre, post=post, tx=tx)
+
+
+@pytest.mark.valid_from("Istanbul")
```

Comment:
Additional tests that follow cover cases found in https://github.com/ethereum/tests/blob/develop/src/GeneralStateTestsFiller/stPreCompiledContracts2/CALLBlake2fFiller.json